### PR TITLE
Sampling of the redshifts according to the Peng et al(2010) model.

### PR DESCRIPTION
**Pull request by philipp128**
## Description
This module samples the redshifts from a Schechter function according to the model of [1].
## Checklist
- [ ] Follow the [Contributor Guidelines](https://github.com/skypyproject/skypy/blob/master/CONTRIBUTING.md)
- [ ] Write unit tests
- [ ] Write documentation strings
- [ ] Assign someone from your working team to review this pull request
- [ ] Assign someone from the infrastructure team to review this pull request

### CODE REVIEW

**Comment by ntessore** [comment]
How is the marginal redshift distribution for the Schechter stellar mass function different from the marginal redshift distribution for the Schechter luminosity function?

**Comment by philipp128** [comment]
> How is the marginal redshift distribution for the Schechter stellar mass function different from the marginal redshift distribution for the Schechter luminosity function?

The redshift dependence of phi_star is different. One could change the sampling in the way that one can use different pdfs but I would like to leave this for later.

File: skypy/galaxy/redshift.py [hunk 1]
```diff
@@ -249,3 +250,160 @@ def herbel_pdf(redshift, alpha, a_phi, b_phi, a_m, b_m, cosmology,
...
+
+    Parameters
+    ----------
+    alpha : float or scalar
+        The alpha parameter in the Schechter stellar mass function
+    a_phi, b_phi : float or scalar
+        Slope and intercept of the linear parametrisation of Phi_*.
+    m_star : float or int
+        Characteristic stellar mass M_*.
+    cosmology : instance
```
Comment (Lucia-Fonseca) [inline skypy/galaxy/redshift.py:268]: is `instance` a type? I usually write here `astropy.cosmology.Cosmology`

File: skypy/galaxy/redshift.py [hunk 2]
```diff
@@ -249,3 +250,160 @@ def herbel_pdf(redshift, alpha, a_phi, b_phi, a_m, b_m, cosmology,
...
+        Instance of an Astropy Cosmology class.
+    low : float or array_like of floats, optional
```
Comment (Lucia-Fonseca) [inline skypy/galaxy/redshift.py:270]: Can you specify that this if the low boundary for redshift? Also, I do not see why we want a default value here

File: skypy/galaxy/redshift.py [hunk 3]
```diff
@@ -249,3 +250,160 @@ def herbel_pdf(redshift, alpha, a_phi, b_phi, a_m, b_m, cosmology,
...
+        The lower boundary of the output interval. All values generated will be
+        greater than or equal to low.
+        The default value is 0.0.
+    high : float or array_like of floats, optional
```
Comment (Lucia-Fonseca) [inline skypy/galaxy/redshift.py:274]: same as low

File: skypy/galaxy/redshift.py [hunk 4]
```diff
@@ -249,3 +250,160 @@ def herbel_pdf(redshift, alpha, a_phi, b_phi, a_m, b_m, cosmology,
...
+    size : int or tuple of ints, optional
+        The number of redshifts to sample. If None one value is sampled.
+    m_min : float or int
+        Lower limit of the stalar mass.
+    resolution : int, optional
+        Characterises the resolution of the sampling. Default is 100.
+
+    Returns
+    -------
+    redshift_sample : ndarray or float
```
Comment (Lucia-Fonseca) [inline skypy/galaxy/redshift.py:286]: I would write:
```suggestion
    redshift_sample : (size, ) array_like
```

File: skypy/galaxy/redshift.py [hunk 5]
```diff
@@ -249,3 +250,160 @@ def herbel_pdf(redshift, alpha, a_phi, b_phi, a_m, b_m, cosmology,
...
+    r"""Calculates the redshift pdf of the Schechter stellar mass function
+    according to the model of Peng et al. [1]_ equation (25).
+
+    That is, multiplying by the comoving element
+    using a flat :math:`\Lambda \mathrm{CDM}` model to get the corresponding
+    pdf.
+
+    Parameters
+    ----------
+    redshift : array_like
```
Comment (Lucia-Fonseca) [inline skypy/galaxy/redshift.py:352]:
```suggestion
    redshift : (nz,) array_like
```

File: skypy/galaxy/redshift.py [hunk 6]
```diff
@@ -249,3 +250,160 @@ def herbel_pdf(redshift, alpha, a_phi, b_phi, a_m, b_m, cosmology,
...
+        Input redshifts.
+    alpha : float or scalar
```
Comment (Lucia-Fonseca) [inline skypy/galaxy/redshift.py:354]: What do you mean by scalar?

File: skypy/galaxy/redshift.py [hunk 7]
```diff
@@ -249,3 +250,160 @@ def herbel_pdf(redshift, alpha, a_phi, b_phi, a_m, b_m, cosmology,
...
+        The alpha parameter in the Schechter stellar mass function
+    m_star : float or int
+        Characteristic stellar mass M_*.
+    m_min : float or int
+        Lower limit of the stalar mass.
+    a_phi, b_phi : float or scalar
```
Comment (Lucia-Fonseca) [inline skypy/galaxy/redshift.py:360]: scalar?

File: skypy/galaxy/redshift.py [hunk 8]
```diff
@@ -249,3 +250,160 @@ def herbel_pdf(redshift, alpha, a_phi, b_phi, a_m, b_m, cosmology,
...
+        Slope and intercept of the linear parametrisation of Phi_*.
+    cosmology : instance
```
Comment (Lucia-Fonseca) [inline skypy/galaxy/redshift.py:362]: instance? see previous function

File: skypy/galaxy/redshift.py [hunk 9]
```diff
@@ -249,3 +250,160 @@ def herbel_pdf(redshift, alpha, a_phi, b_phi, a_m, b_m, cosmology,
...
+        Instance of an Astropy Cosmology class.
+
+    Returns
+    -------
+    pdf : ndarray or float
```
Comment (Lucia-Fonseca) [inline skypy/galaxy/redshift.py:367]:
```suggestion
    pdf : (nz,) array_like
```

File: skypy/galaxy/redshift.py [hunk 10]
```diff
@@ -249,3 +250,160 @@ def herbel_pdf(redshift, alpha, a_phi, b_phi, a_m, b_m, cosmology,
...
+    Un-normalised probability density function as a function of redshift
+    according to Peng et al. [1]_.
```
Comment (Lucia-Fonseca) [inline skypy/galaxy/redshift.py:369]:
```suggestion
    according to Peng et al. [1].
```
You can only use the underscore once for references.

File: skypy/galaxy/redshift.py [hunk 11]
```diff
@@ -249,3 +250,160 @@ def herbel_pdf(redshift, alpha, a_phi, b_phi, a_m, b_m, cosmology,
...
+
+    Notes
+    -----
+    This module calculates the function
```
Comment (Lucia-Fonseca) [inline skypy/galaxy/redshift.py:373]: Equation number?

**Comment by ntessore** [comment]
closing this in favour of the existing model

**Pull request closed without merging**